Fix generation of oneOf interfaces for oneOf usage in properties

bkabrda opened the pull request:
This extends the recently added oneOf support to also respect oneOf in properties, as reported at https://github.com/OpenAPITools/openapi-generator/issues/5382

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

CC @jimschubert who reviewed the original oneOf PR and this is a followup further improving that functionality (I tested this for both Java and go-experimental and both seem to work fine with this).

dummyalias commented:
First of: Thanks a ton for all your effort on this  problem! It will probably save me hours of work and headache.

The new commit seems to introduce a new problem (at least when using the `jersey2` library). When using this specification:
```yaml
openapi: "3.0.2"
info:
    version: 1.0.0
    title: Foo
paths:
    /a:
        get:
            responses:
                '200':
                    description: a
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/A'
components:
    schemas:
        Id:
            type: string
        A:
            type: object
            properties:
                id:
                    allOf:
                        - $ref: '#/components/schemas/Id'
                    description: The id of a
```

This happens:

```console
$ java -jar ~/git/bkabrda/openapi-generator/modules/openapi-generator-cli/target/openapi-generator-cli.jar generate \
   -i openapi.yaml \
   -g java -p 'library=jersey2' \
   -o ./out
[main] INFO  o.o.codegen.DefaultGenerator - Generating with dryRun=false
[main] INFO  o.o.codegen.DefaultGenerator - OpenAPI Generator: java (client)
[main] INFO  o.o.codegen.DefaultGenerator - Generator 'java' is considered stable.
[main] INFO  o.o.c.languages.AbstractJavaCodegen - Environment variable JAVA_POST_PROCESS_FILE not defined so the Java code may not be properly formatted. To define it, try 'export JAVA_POST_PROCESS_FILE="/usr/local/bin/clang-format -i"' (Linux/Mac)
[main] INFO  o.o.c.languages.AbstractJavaCodegen - NOTE: To enable file post-processing, 'enablePostProcessFile' must be set to `true` (--enable-post-process-file for CLI).
[main] INFO  o.o.codegen.DefaultCodegen - Skipped overwriting pom.xml as the file already exists in /home/richard/tmp_1m/./out//pom.xml
[main] INFO  o.o.codegen.DefaultCodegen - Skipped overwriting README.md as the file already exists in /home/richard/tmp_1m/./out//README.md
[main] INFO  o.o.codegen.DefaultCodegen - Skipped overwriting build.gradle as the file already exists in /home/richard/tmp_1m/./out//build.gradle
[main] INFO  o.o.codegen.DefaultCodegen - Skipped overwriting build.sbt as the file already exists in /home/richard/tmp_1m/./out//build.sbt
[main] INFO  o.o.codegen.DefaultCodegen - Skipped overwriting settings.gradle as the file already exists in /home/richard/tmp_1m/./out//settings.gradle
[main] INFO  o.o.codegen.DefaultCodegen - Skipped overwriting gradle.properties as the file already exists in /home/richard/tmp_1m/./out//gradle.properties
[main] INFO  o.o.codegen.DefaultCodegen - Skipped overwriting AndroidManifest.xml as the file already exists in /home/richard/tmp_1m/./out/src/main/AndroidManifest.xml
[main] WARN  o.o.codegen.DefaultCodegen - Empty operationId found for path: GET /a. Renamed to auto-generated operationId: aGET
Exception in thread "main" java.lang.NullPointerException
	at org.openapitools.codegen.DefaultCodegen.addOneOfInterfaceModel(DefaultCodegen.java:5746)
	at org.openapitools.codegen.DefaultCodegen.preprocessOpenAPI(DefaultCodegen.java:752)
	at org.openapitools.codegen.languages.AbstractJavaCodegen.preprocessOpenAPI(AbstractJavaCodegen.java:1071)
	at org.openapitools.codegen.DefaultGenerator.configureGeneratorProperties(DefaultGenerator.java:211)
	at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:993)
	at org.openapitools.codegen.cmd.Generate.run(Generate.java:426)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:61)
```

bkabrda commented:
Hey @dummyalias thanks for catching this. I should be able to find some time until EOW to get that fixed.

bkabrda commented:
@dummyalias thanks for pointing this out. The issue actually only appeared once I rebased on top of latest master, so I guess something must have changed in between. Either way I fixed it and hopefully everything is ok now.

dummyalias commented:
Sorry for the late response, I had a cold and wasn't at work.

The error I described is now gone, and code generation works again, but the code doesn't compile. An interface with the name `AIdOneOf` ist generated. Imports for `JsonTypeInfo` and `JsonSubTypes` are missing in the corresponding java file. This interface should probably not be generated at all.

bkabrda commented:
@dummyalias can you post the actual spec? The sample that you pasted has `allOf`, but you're mentioning `oneOf`, so I guess the one you posted is not entirely accurate?

dummyalias commented:
I'm using the same spec, given above. A `*OneOf` model is generated, even though `oneOf` doesn't appear in the spec.

bkabrda commented:
Oh, I see. Ok, that's definitely a problem then. Will try to debug and fix, thanks for clarifying!

bkabrda commented:
@dummyalias should be fixed now, feel free to re-test. Thanks!

dummyalias commented:
Works a treat now. Thanks again!

staaouat commented:
I just installed the last version (openapi-generator-cli-4.3.0-20200326.174439-234.jar) but  the class  org.openapitools.client.model.OneOfstringstring still broken in model.

wing328 commented:
@staaouat please open an issue with details (minimal spec, expected code, etc) for tracking.

rgomezp commented:
Can someone please look into https://github.com/OpenAPITools/openapi-generator/issues/6161 ? Seems to still be happening
